src/test/java/uk/nhs/hee/tis/trainee/credentials/service/CredemtoalEvemtServiceTest.java

```diff
@@ -0,0 +1,176 @@
+import uk.nhs.hee.tis.trainee.credentials.TestJwtUtil;
+import uk.nhs.hee.tis.trainee.credentials.dto.CredentialType;
+import uk.nhs.hee.tis.trainee.credentials.model.CredentialMetadata;
+import uk.nhs.hee.tis.trainee.credentials.repository.CredentialMetadataRepository;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+class CredentialEventServiceTest {
+
+
+  private static final UUID PLACEMENT_1_credentialId =
+      UUID.fromString("e2a26a50-2c51-4327-8692-1cb05b80af07");
+  private static final String PLACEMENT_1_traineeId = "47165";
+  private static final String PLACEMENT_1_tisId = "2bd3f009-9c7a-4352-bcc6-9dff8eac07d6";
+  private static final Instant PLACEMENT_1_issuedAt =
+      Instant.parse("2023-09-27T09:37:48.000+00:00");
+
+  private static final UUID PLACEMENT_2_credentialId =
+      UUID.fromString("e5c09a44-39ff-442f-b356-8424caa581d0");
+  private static final String PLACEMENT_2_traineeId = "47165";
+  private static final String PLACEMENT_2_tisId = "2bd3f009-9c7a-4352-bcc6-9dff8eac07d6";
+  private static final Instant PLACEMENT_2_issuedAt =
+      Instant.parse("2023-08-14T14:12:54.000+00:00");
+
+  private static final UUID PLACEMENT_3_credentialId =
+      UUID.fromString("6ada9134-a582-44a2-8ce0-106ae8b9bcd4");
+  private static final String PLACEMENT_3_traineeId = "47165";
+  private static final String PLACEMENT_3_tisId =
+      "72055636-1439-11ee-be56-0242ac120002";
+  private static final Instant PLACEMENT_3_issuedAt =
+      Instant.parse("2023-09-27T09:41:35.000+00:00");
+
+  private static final UUID PROGRAMME_1_credentialId =
+      UUID.fromString("6ada9134-a582-44a2-8ee0-125ae8b9bcd4");
+  private static final String PROGRAMME_1_traineeId = "47165";
+  private static final String PROGRAMME_1_tisId = "72055636-1439-11ee-be56-0242ac120002";
+  private static final Instant PROGRAMME_1_issuedAt =
+      Instant.parse("2023-09-27T09:41:35.000+00:00");
+
+  private static final UUID PROGRAMME_2_credentialId =
+      UUID.fromString("6ada9134-a582-44a2-8de0-105ae8b9bcd4");
+  private static final String PROGRAMME_2_traineeId = "47165";
+  private static final String PROGRAMME_2_tisId = "72055636-1439-11ee-be56-0242ac120002";
+  private static final Instant PROGRAMME_2_issuedAt =
+      Instant.parse("2023-08-27T09:41:35.000+00:00");
+
+  CredentialMetadata placement1 = new CredentialMetadata();
+  CredentialMetadata placement2 = new CredentialMetadata();
+  CredentialMetadata placement3 = new CredentialMetadata();
+  CredentialMetadata programme1 = new CredentialMetadata();
+  CredentialMetadata programme2 = new CredentialMetadata();
+
+  private CredentialEventService credentialEventService;
+
+  @Mock
+  CredentialMetadataRepository repository;
+
+  @BeforeEach
+  public void setUp() {
+    repository = mock(CredentialMetadataRepository.class);
```

Comment:
The mock should use one of `@Mock` on the field or initialise with `mock(CredentialMetadataRepository.class`, not both.